src/main/java/seedu/address/model/attributes/Date.java

```diff
@@ -60,6 +60,29 @@ public boolean equals(Object other) {
                 && year == ((Date) other).year);
     }
 
+    /**
+     * Returns true if date is earlier than given date.
+     * @param other A valid date.
+     */
+    public boolean compareTo(Date other) {
```

Comment:
Will be better if you make it such that the class implements Comparable<Date> and implement the standard compareTo that overrides the one in Comparable. That way, we can utilize this method in sorting functions as well.